File: ReverseString.c
// Reversing C-Style String (Last character null)
void reverse (char *val){
	char *tail = val;
	char temp;
	if (str) {
		while(*tail){
			++tail;
		}
		--tail; //last char being null
		while(val < tail){
			temp = *val;
			*val++ = *tail;
			*tail-- = temp;
		}
	}
}

File: ReverseString.java
// Reversing C-Style String (Last character null)
static void reverse (char[] val){
	int head = 0;
	int tail = val.length - 2; //last char being null
	while (head < tail){
		char temp = val[head];
		val[head] = val[tail];
		val[tail] = temp;
		head ++; tail --;
	}
}
